src/ds3.c

```diff
@@ -790,6 +799,24 @@ void ds3_request_set_max_keys(ds3_request* _request, uint32_t max_keys) {
     _set_query_param(_request, "max-keys", max_keys_s);
 }
 
+void ds3_request_set_name(ds3_request* _request, const char* name) {
+    _set_query_param(_request, "name", name);
+}
+
+void ds3_request_set_id(ds3_request* _request, const char* id) {
+    _set_query_param(_request, "id", id);
+}
+
+void ds3_request_set_type(ds3_request* _request, object_type type) {
+    char* type_as_string = _get_object_type(type);
+    _set_query_param(_request, "type", type_as_string);
```

Comment:
If `type_as_string` is `null` do we actually want to set the `type` query param?